### DISCUSSION TITLE
413 body exceeded 1mb limit

### DISCUSSION THREAD

**Discussion by schwamic** [vercel/vercel#4529] [Help] [answered]
If you try to upload an image that is larger than 1mb to a serverless function, I get the following error message: `413 Body exceeded 1mb limit`. I use the free hobby account. Could it be that there are some limits or am I doing something wrong?

---

`vercel": "19.0.1`, `"next": "9.4.0"`

**Comment by coetry** [comment]
Hi @schwamic the limit should be 5mb as documented [here](https://vercel.com/docs/v2/platform/limits?query=413#serverless-function-payload-size-limit). The same limit applies to all account tiers. Can you provide a small reproduction for us to take a look at?

**Reply by schwamic** [reply]
hi @coetry, I make a simple post-request, which contains the information of the image, which works under 1MB and ends in a 413 error if the image is bigger than about 1MB.

#### How to reproduce:
1. Setup a local development with vercel and next.js
2. Add a function under `/api`
3. Open Postman and send a POST request with a binary lager than 1MB
4. Response 413 - Body exceeded 1mb limit

I have also written to the support team, maybe they can take a deeper look at my account.

**Comment by schwamic** [comment, marked as answer]
Vercel does allow a body up to 5MB. The error was caused by Next.js because the upload limit is 1MB by default. The documentation explains how to set the api middleware to allow 5MB: [nextjs/docs](https://nextjs.org/docs/api-routes/api-middlewares#custom-config).

Thank you @paulogdm and @coetry for your help.
